test/proxyservice/proxyservice_error_test.go

```diff
@@ -0,0 +1,129 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	"github.com/stretchr/testify/assert"
+)
+
+const SecretlessImageName = "secretless-broker"
+
+func dockerContainer(imageName string) (types.Container, error) {
+	ctx := context.Background()
+	cli, err := client.NewClientWithOpts(client.FromEnv)
+	if err != nil {
+		return types.Container{}, err
+	}
+
+	cli.NegotiateAPIVersion(ctx)
+
+	containerListOptions := types.ContainerListOptions{
+		All:    true,
+		Latest: true,
+	}
+
+	containers, err := cli.ContainerList(context.Background(), containerListOptions)
+	if err != nil {
+		return types.Container{}, err
+	}
+
+	var brokerContainer types.Container
+	for _, container := range containers {
+		if container.Image != imageName {
+			continue
+		}
+		brokerContainer = container
+	}
+
+	if brokerContainer.ID == "" {
+		return types.Container{},
+			fmt.Errorf("Could not find matching container for image '%s'", imageName)
+	}
+
+	return brokerContainer, nil
+}
+
+func dockerLog(container types.Container) (string, error) {
+	ctx := context.Background()
+	cli, err := client.NewClientWithOpts(client.FromEnv)
+	if err != nil {
+		return "", err
+	}
+
+	cli.NegotiateAPIVersion(ctx)
+
+	logOptions := types.ContainerLogsOptions{
+		Timestamps: false,
+		ShowStdout: true,
+	}
+
+	out, err := cli.ContainerLogs(ctx, container.ID, logOptions)
+	if err != nil {
+		return "", err
+	}
+
+	buf := new(bytes.Buffer)
+	buf.ReadFrom(out)
+	return buf.String(), nil
+}
+
+func TestProxyserviceErrors(t *testing.T) {
+	container, err := dockerContainer(SecretlessImageName)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	dockerLog, err := dockerLog(container)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	// Check exit code
+	t.Run("fatal exit enforced", func(t *testing.T) {
+		assert.Equal(t, container.State, "exited")
+		assert.Contains(t, container.Status, "Exited (1)")
+	})
+
+	// Check HTTP connector error handling
+	t.Run("HTTP errors", func(t *testing.T) {
+		assert.Contains(t, dockerLog,
```

Comment:
It should be possible to combine these error test cases into a table, and run table-driven. Your call.